lcls-twincat-motion/Library/Tests/FB_StatePMPSEnables_Test.TcPOU

```diff
@@ -288,6 +292,136 @@ IF tonTimer.Q OR fbMove.bSetDone THEN
     bOneTestDone := TRUE;
     TEST_FINISHED();
 END_IF
+]]></ST>
+      </Implementation>
+    </Method>
+    <Method Name="TestBoundsAfterFallingIntoUnknownState" Id="{3abf745e-697a-46d1-92e5-75b0ab69c495}">
+      <Declaration><![CDATA[METHOD TestBoundsAfterFallingIntoUnknownState
+VAR_INPUT
+    nTestID: UINT;
+    fStartPosDelta: REAL;
+    fGoalPosDelta: REAL;
+END_VAR
+VAR_INST
+    stBoundsCheckMotionStage: ST_MotionStage;
+    fbBoundsCheckMotionStage: FB_MotionStage;
+    fbStateEnables: FB_StatePMPSEnables;
+    fbMotionStageSetAndMove: FB_MotionStageSetAndMoveHelper;
+    fbFFHWO: FB_HardwareFFOutput := (bAutoReset := TRUE);
+    mcSetPos: MC_SetPosition;
+    bInit: BOOL;
+    bSetPos: BOOL;
+    bCommandedMove: BOOL;
+    nGoalStateIndex: UINT;
+END_VAR
+]]></Declaration>
+      <Implementation>
+        <ST><![CDATA[TEST(CONCAT('TestBoundsAfterFallingIntoUnknownState',UINT_TO_STRING(nTestID)));
+IF nTestCounter <> nTestID THEN
+    RETURN;
+END_IF
+
+// On first pass, set the goal state index to a valid state to initialize the bounds.
+// Then, set the goal state index to an unknown state to ensure the bounds remain
+// active.
+IF NOT bInit THEN
+    nGoalStateIndex := nGoalState;
+    SetEnables(stBoundsCheckMotionStage);
+ELSE
+    nGoalStateIndex := 0;
+END_IF
+
+
+// Run our FB which should enable the real move while within the allowable bounds.
+// Once the position is set out of the bounds, the bounds should persist and prevent
+// further movement away from the goal state.
+fbStateEnables(
+    stMotionStage:=stBoundsCheckMotionStage,
+    astPositionState:=astPositionState,
+    fbFFHWO:=fbFFHWO,
+    bEnable:=TRUE,
+    nGoalStateIndex:=nGoalStateIndex,
+    eStatePMPSStatus:=E_StatePMPSStatus.TRANSITION,
+    bTransitionAuthorized:=TRUE,
+);
+
+// Set position to be out of the goal's range, and try to move away from the goal
+fbMotionStageSetAndMove(
+    bExecute:=bInit,
+    fStartPosition:=astPositionState[nGoalState].fPosition + fStartPosDelta,
+    fGoalPosition:=astPositionState[nGoalState].fPosition + fGoalPosDelta,
+    fVelocity:=1.0,
+    stMotionStage:=stBoundsCheckMotionStage,
+    fDelta:=0.01,
+    bResetDone=>,
+    bSetDone=>,
+    bMotionStarted=>,
+    bMoveDone=>
+);
+
+bInit := TRUE;
+
+fbFFHWO.EvaluateOutput();
+
+// If we've reached the position OR ran out of time to set the position, check the asserts
+IF tonTimer.Q OR fbMotionStageSetAndMove.bMoveDone THEN
+    AssertTrue(
+        tonTimer.Q,
+        'Timeout should have occurred. Move should have been unable to complete.',
```

Comment:
```suggestion
        'Timeout should not have occurred. Move should have been unable to complete.',
```

Reply:
The test case intentionally checks to make sure that it did timeout. E.g., the only reason it wouldn't time out is if it completed the move but completing the move is actually the failure condition. So the timeout should occur.

Reply:
Whoops, I realized this about halfway through reviewing the test but left this comment in anyway